kore/src/Kore/Step/Axiom/EvaluationStrategy.hs

```diff
@@ -76,28 +78,40 @@ definitionEvaluation equations =
         let -- Attempt an equation, pairing it with its result, if applicable.
             attemptEquation equation =
                 Equation.attemptEquation condition term' equation
-                >>= return . Bifunctor.second apply
+                >>= return . Bifunctor.bimap (Option . Just . Min) apply
```

Comment:
I think we should use [`ExceptRT`](https://hackage.haskell.org/package/errors-2.3.0/docs/Data-EitherR.html#t:ExceptRT) here and `Foldable.foldlM` below in place of `iterateUntil`.

Reply:
Thanks, that's exactly what I was looking for!